### ISSUE TITLE
Bug - PDF Printing broken since install of perl 5.38

### ISSUE THREAD

**Issue by tomglx** [znuny/Znuny#521]
### Environment

- Server OS: Fedora 39 64-Bit
- Browser: Firefox 120 on Windows 10
- Znuny version: 6.5.4
- Perl Version: 5.38 subversion 2
- PDF::API2 Versions: 2.0.44 and 2.045

### Expected behavior

When printing a ticket, a correct displayable PDF is produced.

### Actual behavior

When printing a ticket, a PDF file is produced, but cannot be displayed by any viewer. The PDF code seems
to be corrupt.  The PDF header at the start of the document seems to be correct.

### How to reproduce

1. Go to any Ticket view
2. Click on Print
3. Save or open the PDF document
4. Content is not visible. Acrobat Reader gives an error. FoxIT PDF Reader does'nt, but also shows an empty document.

### Additional information

What I've tried so far. With no improvement. 
- Tried 2 different PDF::API2 versions. 
- Translated all occurrences of bad width values in PDF.pm from "," to ".".  To suppress the warnings about illegal numeric
values in comparisons.
- Wrote the produced document to the disk of the server. From PDF.pm. It's identical to the one downloaded via web browser. So the web server seems not be a factor here
- When executing the print process, absolutely no errors are written to the web server error log any more. Before the translation from "," to "." in PDF.pm there were some warnings as mentioned before.

### Screenshots

Message from Adobe Reader
![grafik](https://github.com/znuny/Znuny/assets/14212818/26c987f5-def3-402e-9b92-716a5a0fb4ae)
A corrupt PDF file
[Ticket_10431106_2023-12-07_11 59.pdf](https://github.com/znuny/Znuny/files/13599425/Ticket_10431106_2023-12-07_11.59.pdf)

**Comment by KaiHerlemann** [comment]
@tomglx I had no problems at all displaying the attached PDF file (with the integrated PDF readers of Google Chrome 120.0.6099.62 and Firefox 118.0.1).
Only some parts of the file are blacked out, presumably for privacy reasons.

Did you perhaps upload a different file by mistake?

**Comment by tomglx** [comment]
That's exactly right. But you've also discovered another problem. The
internal readers can display it, the standalone readers not.
Like Acrobat Reader, FoxIT Reader and even okular cannot. The blacked out
parts are another problem, no privacy measure.
If I comment out the Background Properties in PDF/Ticket.pm, they become
visible again. The uploaded file is no mistake.


Am Fr., 8. Dez. 2023 um 12:40 Uhr schrieb Kai Herlemann <
***@***.***>:

**Comment by rkaldung** [comment]
@tomglx How did you installed the PDF::API2 version you've tried? And what were their file locations?

**Comment by tomglx** [comment]
I've realized znuny comes with it's own PDF::API2 Version 2.033. So I've
tried already by uninstalling the OS RPM. Also I removed it from the
cpan-lib directory, so that the newer OS RPM 2.044 was used.
No difference. Seems that the root cause remains in the introduction of
perl 5.38.

Roy Kaldung ***@***.***> schrieb am Sa., 9. Dez. 2023, 08:45:

**Comment by tomglx** [comment]
I suppose that I've solved my problem. The LANG=de_DE.utf8 environment variable was set in the Apache Webserver systemd
service. The perl 5.38 interpreter acts on this as if LC_NUMERIC  is set to the same value. This behaviour was introduced in this version. But according to the docs, it requires "use locale" also to be set. I couldn't find any occurrence of this in the znuny sources. Regardless of this, the interpreter assumes that a comma is the decimal separator. This results in many wrong calculations in and for PDF::API2. And for the znuny layout too. Looks like a perl bug for me.
I removed the LANG setting from the systemd service. This change made all working again.
